### ISSUE TITLE
Listcontrol displays non-existent child elements in the "Create Object"-Dialog

### ISSUE THREAD

**Issue by ChristianGreiner** [AdamsLair/duality#444]
Like the title already says, the listcontrol in the "Create Object"-Dialog displays non-existent child elements of the components.

![foo](http://i.imgur.com/BQdc0ud.png)

**Comment by ilexp** [comment]
Non-obvious mode: The dialog is actually intended to show derived types when opening a type item. However, there appears to be a bug, so this does not happen.

In general, displaying a plus sign when there _could_ be child items and removing it when there don't turn out to be any is fine, as evaluating this for all displayed items could be quite time consuming.
